srcdeps-maven-plugin/src/main/java/org/srcdeps/mvn/plugin/SrcdepsInitMojo.java

```diff
@@ -0,0 +1,516 @@
+         *            the SCM URL to create a new {@link ScmRepository.Builder} for
+         * @return a new {@link ScmRepository.Builder}
+         */
+        public ScmRepository.Builder createRepository(String url) {
+
+            ScmRepository.Builder repoBuilder = ScmRepository.builder();
+            Set<String> selectors = new TreeSet<>();
+
+            Map<String, Integer> groupIdCntMap = urlGroupIdMap.get(url);
+            if (groupIdCntMap.size() == 1) {
+                String groupId = groupIdCntMap.keySet().iterator().next();
+                Set<String> urls = groupIdUrlMap.get(groupId);
+
+                if (urls.size() == 1) {
+                    repoBuilder.id(groupId);
+                    selectors.add(groupId);
+                } else {
+                    Set<Gav> gavs = urlGavMap.get(url);
+                    if (gavs.size() == 1) {
+                        Gav gav = gavs.iterator().next();
+                        repoBuilder.id(groupId + "." + gav.getArtifactId().replace('.', '-'));
+                        selectors.add(gav.getGroupId() + ":" + gav.getArtifactId());
+                    } else {
+                        repoBuilder.id(groupId + ".id" + Math.abs(url.hashCode()));
+                        for (Gav gav : gavs) {
+                            selectors.add(gav.getGroupId() + ":" + gav.getArtifactId());
+                        }
+                    }
+                }
+            } else {
+                /* Find the groupId with the highest count of artifacts and the highest length */
+                String id = null;
+                int cnt = 0;
+                for (Entry<String, Integer> groupIdCnt : groupIdCntMap.entrySet()) {
+                    int idLength = id == null ? 0 : id.length();
+                    int newCnt = groupIdCnt.getValue().intValue();
+                    String newId = groupIdCnt.getKey();
+                    if (cnt < newCnt || (cnt == newCnt && newId.length() > idLength)) {
+                        id = newId;
+                        cnt = newCnt;
+                    }
+                }
+
+                if (urlGroupIdMap.get(url).size() == 1) {
+                    repoBuilder.id(id);
+                } else {
+                    repoBuilder.id(id + ".id" + Math.abs(url.hashCode()));
+                }
+
+                for (String groupId : groupIdCntMap.keySet()) {
+                    Set<String> urls = groupIdUrlMap.get(groupId);
+                    if (urls.size() == 1) {
+                        selectors.add(groupId);
+                    } else {
+                        for (Gav gav : urlGavMap.get(url)) {
+                            selectors.add(gav.getGroupId() + ":" + gav.getArtifactId());
+                        }
+                    }
+                }
+
+            }
+
+            for (String selector : selectors) {
+                repoBuilder.selector(selector);
+            }
+
+            return repoBuilder.url(url);
```

Comment:
I have to admit that this code was a bit of a mystery to me. It all looks good, but I'm missing a bigger picture.